views/layouts/MainLayout.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { Outlet, useLocation, useNavigate } from 'react-router-dom'
+
+import Header from '../components/Header'
+import Sidebar from '../components/Sidebar'
+
+import getAvatar from '../fe-utils/avatarUtil'
+import getCounts from '../fe-utils/countUtil'
+import { fetchSubjects } from '../fe-utils/fetchUtil'
+import { AuthContext } from '../contexts/AuthContext'
+import { headerTitle } from './helpers'
+
+const MainLayout = ({ classes, theme }) => {
+  const [user] = useContext(AuthContext)
+  const { pathname } = useLocation()
+  const navigate = useNavigate()
+
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [sideBarState, setSideBarState] = useState({
+    totalDays: 0,
+    totalStudies: 0,
+    totalSubjects: 0,
+  })
+  const [avatar, setAvatar] = useState('')
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+
+  useEffect(() => {
+    fetchSubjects().then((acl) => {
+      setSideBarState(getCounts({ acl }))
+    })
+    setAvatar(getAvatar({ user }))
```

Comment:
why do we need to set this in state? We should be able to define a constant instead `const avatar = getAvatar({ user })`